chainimport/headers_import.go

```diff
@@ -0,0 +1,1919 @@
+
+	case ForceOverwrite:
+		// Overwrite all headers in overlap region without validation.
+		err := s.overwriteHeaders(region.Start, region.End, options)
+		if err != nil {
+			return fmt.Errorf("overlap overwrite failed: %w", err)
+		}
+		result.ProcessedCount += int(region.End - region.Start + 1)
+		result.AddedCount += int(region.End - region.Start + 1)
+	}
+
+	return nil
+}
+
+func (s *HeadersImport) processDivergenceRegion(ctx context.Context,
+	region HeaderRegion, options ImportOptions, result *ImportResult) error {
+
+	if !region.Exists {
+		return nil
+	}
+
+	log.Infof("Processing divergence region from heights %d to %d for block "+
+		"and filter headers", region.Start, region.End)
+
+	switch options.DivergenceMode {
+	case ForceReconcile:
+		// Force reconcile all headers in divergence region by directly
+		// overwriting both target stores with headers from the authoritative
+		// import source, without performing any validation of existing headers.
+		// This ensures both target stores are synchronized at the same height
+		// with identical data from the import source. This also would help
+		err := s.reconcileHeaders(region.Start, region.End, options)
+		if err != nil {
+			return fmt.Errorf("forced divergent headers reconciliation "+
+				"failed: %w", err)
+		}
+		result.ProcessedCount += int(region.End - region.Start + 1)
+		result.AddedCount += int(region.End - region.Start + 1)
+
+	case ValidateLeadAndSyncLag:
+		// First, validate that the headers in the leading store match the
+		// import source. If they match, preserve those headers in the leading
+		// store and synchronize the lagging store by copying headers from the
+		// import source to fill any gaps. This ensures both stores end up at
+		// the same height without modifying valid headers. If any validation
+		// mismatch is found, the entire import operation is aborted.
+		err := s.validateLeadAndSyncLagTargetStore(
+			ctx, region.Start, region.End, options,
+		)
+		if err != nil {
+			return fmt.Errorf("divergent headers validation failed: %w", err)
+		}
+		result.SkippedCount += int(region.End - region.Start + 1)
+
+	case ValidateAndReconcile:
+		// Compare existing headers in both stores against the import source.
+		// Headers that match the import source are preserved, while any that
+		// differ are overwritten with data from the import source. This ensures
+		// both stores are synchronized with accurate data while minimizing
+		// unnecessary writes. The function returns the count of headers that
+		// needed reconciliation.
+		reconciled, err := s.validateAndReconcileHeaders(
+			region.Start, region.End, options,
+		)
+		if err != nil {
+			return fmt.Errorf("divergent headers reconciliation "+
+				"failed: %w", err)
+		}
+		result.ProcessedCount += int(region.End - region.Start + 1)
+		result.AddedCount += reconciled
+		result.SkippedCount += int(region.End-region.Start+1) - reconciled
+
+	case AbortIfDiverged:
```

Comment:
Same here re complexity: What are we trying to achieve with the headers import?
The original goal that I remember being discussed (might have changed in the meantime, so I'm genuinely asking) was: Have a way to pre-load headers into the database to cut down on the initial sync time massively.
So it would just be a one-time thing, and only if the database is completely empty.

Which means: We wouldn't really need to handle all these divergence cases. Either the DB is empty and we pre-load it, or it isn't and we don't do anything.

Reply:
> So it would just be a one-time thing, and only if the database is completely empty.

The way I've been approaching it, we want to allow for minimal dynamic logic on the behalf of the clients integrating something like this. So a mobile app would just point to the latest header import file of the app author, and let `lnd` handle the rest. 

If `neutrino` doesn't handle the issue of divergent file vs disk information, then the wallet author or user (or `lnd` wrapper) would need to handle that. 

Testnet 3+4 would be a good test bed for this type of logic, as large re-orgs are very frequent.